### ISSUE TITLE
Refactor score/score_dl and select/select_dl to a common interface

### ISSUE THREAD

**Issue by goergen95** [goergen95/seapig#55]
Currently, we support two different interfaces for each scoring and selecting. These are the `select()` and `select_dl()` methods and the `score()` and `score_dl()` methods. These should be refactored to a common interface for scoring and selecting in the form of the `select()` and `score()` method. Each of those should support that query embeddings/logits are supplied directly or via the model/dataloader pattern. However, these options should not be mixed in a single call.
